reports-backend/src/main/java/org/jboss/da/listings/impl/service/ArtifactServiceImpl.java

```diff
@@ -35,7 +35,7 @@ public boolean isArtifactPresent(GAV gav) {
 
     @Override
     public boolean removeArtifact(String groupId, String artifactId, String version) {
-        T artifact = getDAO().findArtifact(groupId, artifactId, version);
+        T artifact = getDAO().findArtifact(groupId, artifactId, version).get();
```

Comment:
Get will throw NoSuchElementException when it is empty. In the following if, artifact can't be null.